### PR TITLE
[fix](statistics)Escape fetch partition stats sql.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Sql to fetch partition stats need to be escaped.

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java [hunk 1]
```diff
@@ -144,12 +144,12 @@ public static List<ResultRow> queryColumnStatisticsByPartitions(TableIf table, S
...
         StringJoiner sj = new StringJoiner(",");
         for (String colName : columnNames) {
-            sj.add("'" + colName + "'");
+            sj.add("'" + StatisticsUtil.escapeSQL(colName) + "'");
         }
         params.put("columnInfo", sj.toString());
         sj = new StringJoiner(",");
         for (String part : partitionNames) {
-            sj.add("'" + part + "'");
+            sj.add("'" + StatisticsUtil.escapeSQL(part) + "'");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java:152]: add test cases?

**Comment by morrySnow** [comment]
run feu't

**Comment by morrySnow** [comment]
run feut

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java (hunk 1 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsRepository.java:152]: added

**Comment by Jibing-Li** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Comment by Jibing-Li** [comment]
run external

**Pull request merged**